source/concepts/applications.rst

```diff
@@ -0,0 +1,358 @@
+  a quaternion.
+
+.. note::
+
+   The avatar description currently only support VR controllers. See `issue #97 in
+   nanover-protocol <https://github.com/IRL2/nanover-protocol/issues/97>`_ for
+   hand-tracking support.
+
+How to represent the avatar is the responsibility of the client. It must assume
+that any of the information may be missing.
+
+In summary, an avatar is structured as such:
+
+.. code::
+
+   avatar.<PLAYER_ID>: {
+     components : [
+       {
+         name
+         position: 
+         rotation
+       }
+     ],
+     playerid,
+     name,
+     color,
+   }
+
+.. _user-origin-description:
+
+User origin
+~~~~~~~~~~~
+
+Avatars are shared in server space. Each client is responsible to locate its
+server space wherever it prefers relative to its game space. However, the
+server can suggest how to center the game space relative to the server space in
+order to place the users according to each other. This is used by the
+:ref:`radial orient <radial-orient>` server feature.
+
+.. note::
+
+   We assume the user origin is always provided by the server. However, it can
+   come from a client so it is possible to implement a client that will place
+   the users relative to each other folowwing arbitrary pattern. This can be
+   used, for instance, to prototype alternative to the radial orient feature
+   without mofifying the server.
+
+The suggested user origin describes where the server suggests a given user
+places the center of its game space and how to rotate that space. The
+origin is described as a protobuf Struct under the key
+``user-origin.<PLAYER_ID>`` where ``<PLAYER_ID>`` is the ID of the user to whom
+the suggestion is addressed. The Struct has the following keys:
+
+* ``position`` is the suggested location of the center for the user's game
+  space in the server space;
+* ``rotation`` is a quaternion describing the rotation of the user's game
+  space in the server space.
+
+A client may not follow the server suggestion and should not be assumed to do
+so. If the key is absent from the shared state, the client may locate itself in
+the server space as it chooses.
+
+.. warning::
+
+   A client has no way of knowing if the user origin emanates from a ligitimate
+   source (i.e. the server or a trusted client). Therefore, a client could use
+   this feature to move users without their conscent. This could cause
+   discomfort if not used responsibly.
+
+As a summary, the user origin is specified as follow in the shared state:
+
+.. code::
+
+   user-origin.<PLAYER_ID>: {
+     position,
+     rotation,
+   }
+
+.. _play-space-description:
+
+Play space
+~~~~~~~~~~
+
+A client can share the shape of its play space with the others. A play space,
+or play area, is the area the user can safely reach. This is mostly relevant
+for VR clients which have to define such a safe space.
+
+The play area is defined as four points, each as a vector of 3 XYZ values, in
```

Comment:
@Ragzouken I think I got the play space correct but it is very different from the rest so I would not mind you conforming this doc is right.

Reply:
Seems right: we take the play area to be a rectangle, take the four corner positions and transform them into server space